pkg/registries/quay/quay.go

```diff
@@ -148,8 +148,11 @@ func (q *Quay) Test() error {
 	}
 	resp, err := client.Get(discoveryURL)
 	if err != nil {
-		log.Errorf("Quay error response: %d", resp.StatusCode)
-		return errors.Errorf("received http status code %d from Quay. Check Central logs for full error.", resp.StatusCode)
+		log.Errorf("Quay error response: %s", err)
+		if resp == nil {
```

Comment:
I think this is always the case, but I won't block the PR over this

Reply:
If `err == nil`, then `resp != nil` is guaranteed. But the opposite is not true: If `err != nil`, then `resp` may be `nil`. This would happen for any error that happened before a connection has been made - in this case there can be no response.